### ISSUE TITLE
Make sure pdf/doc files etc are indexed correctly via the JSON v2 connector

### ISSUE THREAD

**Issue by code-geek** [NASA-IMPACT/sde-indexing-helper#628]
## Description
I got it working by mapping the fileext to this value pattern : ParseFromLastSep($.url_list.url, ".") defaulting to htm when empty (edited) 

this is going to break in most cases though

because for a URL like https://power.larc.nasa.gov/static/publications/, the result of that value pattern will be gov/static/publications/

one option is to first parse from the last / , then from the last . and again defaulting to htm but I'll leave the details to you (edited) 

another detail of note

I don't remember what it looked like yesterday, but I see today the $.url_list.title value from the API is coming back with the fileext already included if the file is a pdf

that's fine, but I recommend making that a conscious decision and adhering to it as convention

because I was initially using this pattern for the filename: Concat($.url_list.title, fileext) which would break if the title already had the extension in it

but would work if it didn't, in the case of real HTML webpages, for example

another recommendation regarding mappings the JSON connector does not do automatically for you

the URL you go to when you click on a search result is whatever is mapped to URL1

so right now, if you click a search result for the JSON test collection, you go nowhere

you just have to add a mapping for URL1 like you would for id $.url_list.url

in summary, do the following:
map fileext to the file at the source (pdf, htm, etc)
map filename to a filename that matches the extension (usually title+fileext)
remove the file extension filters you have in the XML right now
and I recommend this additional step:
map URL1 so you can link out to the original source page

also keep in mind you may need to recycle the app pools on IIS in order to flush cached previews when testing this
